configure

```diff
@@ -1,779 +1,359 @@
-# $CFG_USING_CLANG rather than $CFG_ENABLE_CLANG.
-
-# Set CFG_{CC,CXX,CPP,CFLAGS,CXXFLAGS,LDFLAGS}
-envopt CC
-envopt CXX
-envopt CPP
-envopt CFLAGS
-envopt CXXFLAGS
-envopt LDFLAGS
-
-# a little post-processing of various config values
-CFG_PREFIX=${CFG_PREFIX%/}
-CFG_MANDIR=${CFG_MANDIR%/}
-CFG_DOCDIR=${CFG_DOCDIR%/}
-CFG_BINDIR=${CFG_BINDIR%/}
-CFG_HOST="$(echo $CFG_HOST | tr ',' ' ')"
-CFG_TARGET="$(echo $CFG_TARGET | tr ',' ' ')"
-
-# copy build-triples to host-triples so that builds are a subset of hosts
-V_TEMP=""
-for i in $CFG_BUILD $CFG_HOST;
-do
-   echo "$V_TEMP" | grep -qF $i || V_TEMP="$V_TEMP${V_TEMP:+ }$i"
-done
-CFG_HOST=$V_TEMP
-
-# copy host-triples to target-triples so that hosts are a subset of targets
-V_TEMP=""
-for i in $CFG_HOST $CFG_TARGET;
-do
-   echo "$V_TEMP" | grep -qF $i || V_TEMP="$V_TEMP${V_TEMP:+ }$i"
-done
-CFG_TARGET=$V_TEMP
-
-step_msg "writing configuration"
-
-putvar CFG_SRC_DIR
-putvar CFG_SRC_DIR_RELATIVE
-putvar CFG_BUILD_DIR
-putvar CFG_OSTYPE
-putvar CFG_CPUTYPE
-putvar CFG_CONFIGURE_ARGS
-putvar CFG_PREFIX
-putvar CFG_HOST
-putvar CFG_TARGET
-putvar CFG_LIBDIR_RELATIVE
-putvar CFG_DISABLE_MANAGE_SUBMODULES
-putvar CFG_AARCH64_LINUX_ANDROID_NDK
-putvar CFG_ARM_LINUX_ANDROIDEABI_NDK
-putvar CFG_ARMV7_LINUX_ANDROIDEABI_NDK
-putvar CFG_I686_LINUX_ANDROID_NDK
-putvar CFG_X86_64_LINUX_ANDROID_NDK
-putvar CFG_NACL_CROSS_PATH
-putvar CFG_MANDIR
-putvar CFG_DOCDIR
-putvar CFG_BINDIR
-putvar CFG_USING_LIBCPP
-
-msg
-copy_if_changed ${CFG_SRC_DIR}src/bootstrap/mk/Makefile.in ./Makefile
-move_if_changed config.tmp config.mk
-rm -f config.tmp
-touch config.stamp
-
-if [ -z "$CFG_ENABLE_DEBUG" ]; then
-    step_msg "configured in release mode. for development consider --enable-debug"
-else
-    step_msg "complete"
-fi
-
-if [ "$CFG_SRC_DIR" = `pwd` ]; then
-    X_PY=x.py
-else
-    X_PY=${CFG_SRC_DIR_RELATIVE}x.py
-fi
-
-msg "run \`python ${X_PY} --help\`"
-msg
+o("manage-submodules", "build.submodules", "let the build manage the git submodules")
+o("jemalloc", "rust.use-jemalloc", "build liballoc with jemalloc")
+o("full-bootstrap", "build.full-bootstrap", "build three compilers instead of two")
+o("extended", "build.extended", "build an extended rust tool set")
+
+v("build", "build.build", "GNUs ./configure syntax LLVM build triple")
+v("host", None, "GNUs ./configure syntax LLVM host triples")
+v("target", None, "GNUs ./configure syntax LLVM target triples")
+
+v("set", None, "set arbitrary key/value pairs in TOML configuration")
+
+def p(msg):
+    print("configure: " + msg)
+
+def err(msg):
+    print("configure: error: " + msg)
+    sys.exit(1)
+
+if '--help' in sys.argv or '-h' in sys.argv:
+    print('PRINT HELP')
+    sys.exit(0)
+
+# Parse all command line arguments into one of these three lists, handling
+# boolean and value-based options separately
+unknown_args = []
+need_value_args = []
+known_args = {}
+
+p("processing command line")
+i = 1
+while i < len(sys.argv):
```

Comment:
Could we use [`argparse`](https://docs.python.org/3/library/argparse.html#module-argparse) (or [`optparse`](https://docs.python.org/3/library/optparse.html) if we care about supporting the dead hand of Python 2.6) rather than rolling our own?

Reply:
It's possible, yes, but I'm pretty unfamiliar with both (I barely know python) and would prefer to focus on compatibility for now